src/Grpc.Net.Common/Compression/DeflateCompressionProvider.cs

```diff
@@ -0,0 +1,78 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+#if NET6_0_OR_GREATER
+using System.IO;
+using System.IO.Compression;
+
+namespace Grpc.Net.Compression
+{
+    /// <summary>
+    /// Deflate compression provider.
+    /// </summary>
+    public class DeflateCompressionProvider : ICompressionProvider
```

Comment:
```suggestion
    public sealed class DeflateCompressionProvider : ICompressionProvider
```
With more and better support in the JIT for devirtualization `sealed` becomes important (again).

And "un-sealing" a type won't be a breaking change. The other way around, to seal an unsealed type, would be a breaking change.

Reply:
Good point.

There is one other compression provider and it isn't sealed. I'll make it sealed as well. It's breaking, but no one has a reason to inherit from it.